awx/main/dispatch/pool.py

```diff
@@ -354,6 +355,20 @@ def cleanup(self):
             idx = random.choice(range(len(self.workers)))
             self.write(idx, m)
 
+        # if the database says a job is running on this node, but it's *not*,
```

Comment:
If the database goes away during a job run, `handle_work_error` _does_ get called, but the code in it *also* fails, because it can't reach the database to do `status='failed'` for the appropriate job.  In 3.3.0 and prior, the periodic reaper in the task manager discovered jobs like this.  That implementation was complicated, though, because it means that the task manager had to talk to *all* nodes over AMQP to discover their state.

In this version, every time the dispatcher gets a heartbeat message (every minute or so), it looks to see if there any orphaned tasks for _its_ `execution_node`.  If so, it cleans up after itself.